mnefun/misc.py

```diff
@@ -10,33 +10,37 @@ def make_montage(info, kind, check=False):
     from . import _reorder
     assert kind in ('mgh60', 'mgh70', 'uw_70', 'uw_60')
     picks = mne.pick_types(info, meg=False, eeg=True, exclude=())
+    sphere = mne.make_sphere_model('auto', 'auto', info)
+    info = mne.pick_info(info, picks)
+    to_names = info['ch_names']
     if kind in ('mgh60', 'mgh70'):
-        ch_names = mne.utils._clean_names(
-            [info['ch_names'][pick] for pick in picks], remove_whitespace=True)
         if kind == 'mgh60':
-            assert len(ch_names) in (59, 60)
+            assert len(to_names) in (59, 60)
         else:
-            assert len(ch_names) in (70,)
-        montage = mne.channels.read_montage(kind, ch_names=ch_names)
+            assert len(to_names) in (70,)
+        montage = mne.channels.make_standard_montage(
+            kind, head_size=sphere.radius)
+        from_names = mne.utils._clean_names(to_names, remove_whitespace=True)
     else:
-        ch_names = getattr(_reorder, 'ch_names_' + kind)
-        ch_names = ch_names
-        montage = mne.channels.read_montage('standard_1020', ch_names=ch_names)
-        assert len(montage.ch_names) == len(ch_names)
-        montage.ch_names = ['EEG%03d' % ii for ii in range(1, 61)]
-    sphere = mne.make_sphere_model('auto', 'auto', info)
-    montage.pos /= np.linalg.norm(montage.pos, axis=-1, keepdims=True)
-    montage.pos *= sphere.radius
-    montage.pos += sphere['r0']
-    info = mne.pick_info(info, picks)
+        assert len(to_names) == 60
+        from_names = getattr(_reorder, 'ch_names_' + kind)
+        montage = mne.channels.make_standard_montage(
+            'standard_1020', head_size=sphere.radius)
+    assert len(from_names) == len(to_names)
+    montage_pos = montage._get_ch_pos()
+    montage = mne.channels.make_dig_montage(
```

Comment:
this change requires 0.19 right? Is that assumed / asserted / mentioned somewhere? (sorry if the answer is obvious; this is the first time I'm dabbling with mnefun in several years)

Reply:
The general assumption is that people using `mnefun` are on a (near-)latest `master` version because often we have to fix bugs. So 0.19 should be a safe enough assumption. `make_dig_montage` didn't exist before then so it will just bomb if someone tries this on 0.18.

Additionally also this function isn't really used in `mnefun` analysis scripts by users, it's more of an expert `dev` option for testing new algorithms. So should be okay